src/components/forms/formr-part-b/viewSections/ViewSectionTestHelper.tsx

```diff
@@ -0,0 +1,87 @@
+import { mount } from "@cypress/react";
+import { FormRPartB, Work, Declaration } from "../../../../models/FormRPartB";
+import { DateUtilities } from "../../../../utilities/DateUtilities";
+import { BooleanUtilities } from "../../../../utilities/BooleanUtilities";
+
+type FormRPartBField = keyof FormRPartB;
+type FormRPartBWorkField = keyof Work;
+type FormRPartBDeclarationField = keyof Declaration;
+
+export interface ISectionDataField {
+  fieldName: FormRPartBField;
+  format: string;
+}
+
+export interface ISectionWorkDataField {
+  fieldName: FormRPartBWorkField;
+  format: string;
+}
+
+export interface ISectionDeclarationDataField {
+  fieldName: FormRPartBDeclarationField;
+  format: string;
+}
+
+function CheckDataIsDisplayed(dataValue: any, format: string) {
+  if (typeof dataValue !== "undefined" && dataValue) {
+    if (format === "LocalDate") {
+      const formattedDate = DateUtilities.ToLocalDate(dataValue.toString());
+      cy.get(".nhsuk-summary-list__value").should(
+        "include.text",
+        formattedDate
+      );
+    } else if (format === "YesNo") {
+      const formattedBoolean = BooleanUtilities.ToYesNo(dataValue);
+      cy.get(".nhsuk-summary-list__value").should(
+        "include.text",
+        formattedBoolean
+      );
+    } else {
+      cy.get(".nhsuk-summary-list__value").should("include.text", dataValue);
+    }
+  }
+}
+
```

Comment:
😵‍💫 not a fan of the below, but I might need some help fixing this

Reply:
The issue I ran into, is that I can't reference `formData[formDataItem.fieldName]` if it is possible that `formDataItem.fieldName` is not a property of formData. Even with the `if (formDataItem.fieldName in formData) {` check, the IDE was not happy. I think there's some subtlety around the loss of typing in the conversion to 'raw' javascript that I don't really understand.

Reply:
'Resolved' by using a type of `:any` for the formData object. Some other refactoring has been done, which is a bit of an improvement, but might still be a way to go.